x-pack/test/api_integration/deployment_agnostic/apis/observability/alerting/custom_threshold/consumers.ts

```diff
@@ -0,0 +1,683 @@
+
+      it('should find the created rule with correct information about the consumer', async () => {
+        const match = await alertingApi.findInRules(roleAuthc, ruleId);
+        expect(match).not.to.be(undefined);
+        expect(match.consumer).to.be(consumer);
+      });
+
+      it('should be active and visible from admin role', async () => {
+        const executionStatus = await alertingApi.waitForRuleStatus({
+          roleAuthc,
+          ruleId,
+          expectedStatus: 'active',
+        });
+        expect(executionStatus).to.be('active');
+      });
+
+      it('should be active and visible from infra only role', async () => {
+        await samlAuth.setCustomRole(ROLES.infra_only);
+
+        const infraOnlyRole = await samlAuth.createM2mApiKeyWithCustomRoleScope();
+        const executionStatus = await alertingApi.waitForRuleStatus({
+          roleAuthc: infraOnlyRole,
+          ruleId,
+          expectedStatus: 'active',
+        });
+        expect(executionStatus).to.be('active');
+      });
+
+      it('should NOT be visible from logs only role', async () => {
+        await samlAuth.setCustomRole(ROLES.logs_only);
+        const logsOnlyRole = await samlAuth.createM2mApiKeyWithCustomRoleScope();
+        try {
+          await alertingApi.waitForRuleStatus({
+            roleAuthc: logsOnlyRole,
+            ruleId,
+            expectedStatus: 'active',
+            timeout: 1000 * 3,
+          });
+          throw new Error('Expected rule to not be visible, but it was visible');
+        } catch (error) {
+          expect(error.message).to.contain('timeout');
+        }
+      });
+
+      it('should NOT visible from synthetics only role', async () => {
+        await samlAuth.setCustomRole(ROLES.synthetics_only);
+        const syntheticsOnlyRole = await samlAuth.createM2mApiKeyWithCustomRoleScope();
+        try {
+          await alertingApi.waitForRuleStatus({
+            roleAuthc: syntheticsOnlyRole,
+            ruleId,
+            expectedStatus: 'active',
+            timeout: 1000 * 3,
+          });
+          throw new Error('Expected rule to not be visible, but it was visible');
+        } catch (error) {
+          expect(error.message).to.contain('timeout');
+        }
+      });
+
+      it('should NOT visible from stack alerts only role', async () => {
+        await samlAuth.setCustomRole(ROLES.stack_alerts_only);
+        const stackAlertsOnlyRole = await samlAuth.createM2mApiKeyWithCustomRoleScope();
+        try {
+          await alertingApi.waitForRuleStatus({
+            roleAuthc: stackAlertsOnlyRole,
+            ruleId,
+            expectedStatus: 'active',
+            timeout: 1000 * 3,
+          });
+          throw new Error('Expected rule to not be visible, but it was visible');
+        } catch (error) {
+          expect(error.message).to.contain('timeout');
+        }
+      });
+    });
+
+    describe('Custom threshold - Rule execution - consumer stackAlerts', () => {
```

Comment:
As we move towards allowing `stackAlerts` to be a valid consumer for all observability rules in serverless, these tests will need to be modified a bit.